### PR TITLE
va-segmented-progress-bar - staging review accessibility

### PR DESCRIPTION
**Pull request by powellkerry**
## Chromatic

https://59934-step-indicator-stage-review--60f9b557105290003b387cd5.chromatic.com

## Description
- Add prop/ability to set a custom header level
- Add prop/ability to set the term used to on screen readers to indicate progress (step, chapter, location, etc)
- Removes the aria-label from the div, as this will not get read and the property is confusing not that we have added the labels property
- Add `aria-current="step"` to the current step
- Add `aria-hidden="true"` to the `ol` when there are no labels

Closes https://app.zenhub.com/workspaces/platform-design-system-5f8de67192551b0012ebb802/issues/gh/department-of-veterans-affairs/va.gov-team/59934

## Testing done
- Local testing in Chrome, Mac Voiceover

## Screenshots
- No visual changes (screen readers only)

## Acceptance criteria
- [ ] Meets the requirements of the staging review, voiceover reads the component correctly

## Definition of done
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)

### CODE REVIEW

File: packages/web-components/src/components/va-segmented-progress-bar/va-segmented-progress-bar.tsx [hunk 1]
```diff
@@ -84,14 +94,17 @@ export class VaSegmentedProgressBar {
...
+    const { current, total, label = `Step ${current} of ${total}`, uswds, labels, centeredLabels, counters, headingText, headerLevel, progressTerm } = this;
     let labelsArray;
     if (labels) {
       labelsArray = labels.split(';');
     }
     const range = Array.from({ length: total }, (_, i) => i);
 
     if (uswds) {
+
+      const Tag = `h${headerLevel}`;
```
Comment (ataker) [inline packages/web-components/src/components/va-segmented-progress-bar/va-segmented-progress-bar.tsx:106]: **Must 1:** add header level option ✅

File: packages/web-components/src/components/va-segmented-progress-bar/va-segmented-progress-bar.tsx [hunk 2]
```diff
@@ -128,16 +141,18 @@ export class VaSegmentedProgressBar {
...
-                      <span class="usa-step-indicator__total-steps"> of {total}</span>
-                    </span>
-                    <span class="usa-step-indicator__heading-text">{labels ? labelsArray[current - 1] : headingText}</span>
-                  </h4>
-                </div>
+                {
+                  <div class="usa-step-indicator__header">
+                    <Tag class="usa-step-indicator__heading">
+                      <span class="usa-step-indicator__heading-counter">
+                        <span class="usa-sr-only">{progressTerm}</span>
```
Comment (ataker) [inline packages/web-components/src/components/va-segmented-progress-bar/va-segmented-progress-bar.tsx:148]: **Consider 2:** Add customizable progress term ✅

File: packages/web-components/src/components/va-segmented-progress-bar/va-segmented-progress-bar.tsx [hunk 3]
```diff
@@ -109,10 +122,10 @@ export class VaSegmentedProgressBar {
 
       return (
         <Host>
-          <div class={indicatorClass} aria-label={label}>
```
Comment (ataker) [inline packages/web-components/src/components/va-segmented-progress-bar/va-segmented-progress-bar.tsx:112]: **Should 1:** Remove aria-label from `usa-step-indicator` as it's a non-semantic element ✅

File: packages/web-components/src/components/va-segmented-progress-bar/va-segmented-progress-bar.tsx [hunk 4]
```diff
@@ -109,10 +122,10 @@ export class VaSegmentedProgressBar {
...
-                <ol class="usa-step-indicator__segments">
+          <div class={indicatorClass}>
+                <ol class="usa-step-indicator__segments" aria-hidden={labels ? null : 'true'}>
                   {range.map(step => (
-                    <li class={computeSegmentClass(step)}>
+                    <li class={computeSegmentClass(step)} aria-current={current===step + 1 ? 'step' : null}>
```
Comment (ataker) [inline packages/web-components/src/components/va-segmented-progress-bar/va-segmented-progress-bar.tsx:128]: **Must 2:** add `aria-current` to the current item on the list ✅

File: packages/web-components/src/components/va-segmented-progress-bar/va-segmented-progress-bar.tsx [hunk 5]
```diff
@@ -109,10 +122,10 @@ export class VaSegmentedProgressBar {
 
       return (
         <Host>
-          <div class={indicatorClass} aria-label={label}>
-                <ol class="usa-step-indicator__segments">
+          <div class={indicatorClass}>
+                <ol class="usa-step-indicator__segments" aria-hidden={labels ? null : 'true'}>
```
Comment (ataker) [inline packages/web-components/src/components/va-segmented-progress-bar/va-segmented-progress-bar.tsx:126]: **Must 3:** add `aria-hidden` if labels aren't being used ✅

**Review by ataker — APPROVED** [review]
Note: **Consider 1** was skipped as noted in the original ticket.

File: packages/web-components/src/components/va-segmented-progress-bar/va-segmented-progress-bar.tsx [hunk 6]
```diff
@@ -84,14 +94,17 @@ export class VaSegmentedProgressBar {
   }
 
   render() {
-    const { current, total, label = `Step ${current} of ${total}`, uswds, labels, centeredLabels, counters, headingText } = this;
+    const { current, total, label = `Step ${current} of ${total}`, uswds, labels, centeredLabels, counters, headingText, headerLevel, progressTerm } = this;
```
Comment (micahchiang) [inline packages/web-components/src/components/va-segmented-progress-bar/va-segmented-progress-bar.tsx:97]: huh, TIL you can assign template strings while destructuring.

**Review by briandeconinck — APPROVED** [review]

**Pull request merged**